apps/web/pages/signup.tsx

```diff
@@ -92,6 +93,8 @@ function UsernameField({
     if (formState.isSubmitting || formState.isSubmitSuccessful) return;
 
     async function checkUsername() {
+      // If the username can't be changed, there is no point in doing the username availability check
+      if (disabled) return;
```

Comment:
This also avoids disabling the Signup button for the username that is reserved by the user signing up.